src/Foundatio/Jobs/WorkItemJob/WorkItemJob.cs

```diff
@@ -39,65 +39,58 @@ public class WorkItemJob : IQueueJob, IHaveLogger {
-                return JobResult.FailedWithMessage($"Could not resolve {workItemDataType.Name} work item data type.");
+                return JobResult.FailedWithMessage($"Abandoning {queueEntry.Value.Type} work item: {queueEntry.Id}: Could not resolve work item data type.");
             }
 
             object workItemData;
             try {
                 workItemData = await _queue.Serializer.DeserializeAsync(queueEntry.Value.Data, workItemDataType).AnyContext();
             } catch (Exception ex) {
                 await queueEntry.AbandonAsync().AnyContext();
-                return JobResult.FromException(ex, $"Failed to parse {workItemDataType.Name} work item data.");
+                return JobResult.FromException(ex, $"Abandoning {queueEntry.Value.Type} work item: {queueEntry.Id}: Failed to parse {workItemDataType.Name} work item data.");
             }
 
             var handler = _handlers.GetHandler(workItemDataType);
             if (handler == null) {
                 await queueEntry.CompleteAsync().AnyContext();
-                return JobResult.FailedWithMessage($"Handler for type {workItemDataType.Name} not registered.");
+                return JobResult.FailedWithMessage($"Completing {queueEntry.Value.Type} work item: {queueEntry.Id}: Handler for type {workItemDataType.Name} not registered.");
             }
 
             if (queueEntry.Value.SendProgressReports)
-                await _messageBus.PublishAsync(new WorkItemStatus {
-                    WorkItemId = queueEntry.Value.WorkItemId,
-                    Progress = 0,
-                    Type = queueEntry.Value.Type
-                }).AnyContext();
+                await ReportProgress(handler, queueEntry).AnyContext();
 
             var lockValue = await handler.GetWorkItemLockAsync(workItemData, cancellationToken).AnyContext();
             if (lockValue == null) {
+                handler.Log.Info($"Abandoning {queueEntry.Value.Type} work item: {queueEntry.Id}: Unable to acquire work item lock.");
                 await queueEntry.AbandonAsync().AnyContext();
-                handler.Log.Trace("Unable to acquire work item lock.");
                 return JobResult.Success;
             }
 
             var progressCallback = new Func<int, string, Task>(async (progress, message) => {
-                if (handler.AutoRenewLockOnProgress)
-                    await queueEntry.RenewLockAsync().AnyContext();
-
-                if (handler.AutoRenewLockOnProgress)
-                    await lockValue.RenewAsync().AnyContext();
+                if (handler.AutoRenewLockOnProgress) {
+                    try {
+                        await queueEntry.RenewLockAsync().AnyContext();
+                        await lockValue.RenewAsync().AnyContext();
+                    } catch (Exception ex) {
+                        handler.Log.Error(ex, "Error renewing work item locks: {0}", ex.Message);
+                    }
```

Comment:
Should renewing locks be on a retry as well? I'm wondering if these should be passed a cancellation token?